internal/runtimehandlerhooks/high_performance_hooks.go

```diff
@@ -356,6 +417,171 @@ func setCPUQuota(cpuMountPoint, parentDir string, c *oci.Container, enable bool)
 	return nil
 }
 
+// setCPUPMQOSResumeLatency sets the pm_qos_resume_latency_us for a cpu and stores the original
+// value so it can be restored later. If the latency is an empty string, the original latency
+// value is restored.
+func setCPUPMQOSResumeLatency(c *oci.Container, latency, cpuDir, cpuSaveDir string) error {
+	lspec := c.Spec().Linux
+	if lspec == nil ||
+		lspec.Resources == nil ||
+		lspec.Resources.CPU == nil ||
+		lspec.Resources.CPU.Cpus == "" {
+		return errors.Errorf("find container %s CPUs", c.ID())
+	}
+
+	cpus, err := cpuset.Parse(lspec.Resources.CPU.Cpus)
+	if err != nil {
+		return err
+	}
+
+	for _, cpu := range cpus.ToSlice() {
+		if latency != "" {
+			// retrieve the current latency
+			latencyFile := fmt.Sprintf("%s/cpu%d/power/pm_qos_resume_latency_us", cpuDir, cpu)
+			latencyOrig, err := ioutil.ReadFile(latencyFile)
+			if err != nil {
+				return err
+			}
+
+			// save the current latency so we can restore it later
+			cpuPowerSaveDir := fmt.Sprintf("%s/cpu%d/power", cpuSaveDir, cpu)
+			err = os.MkdirAll(cpuPowerSaveDir, 0o750)
+			if err != nil {
+				return err
+			}
+			latencyFileOrig := fmt.Sprintf("%s/pm_qos_resume_latency_us", cpuPowerSaveDir)
+			err = ioutil.WriteFile(latencyFileOrig, latencyOrig, 0o644)
+			if err != nil {
+				return err
+			}
+
+			// update the pm_qos_resume_latency_us
+			err = ioutil.WriteFile(latencyFile, []byte(latency), 0o644)
+			if err != nil {
+				return err
+			}
+		} else {
+			// retrieve the original latency
+			latencyFileOrig := fmt.Sprintf("%s/cpu%d/power/pm_qos_resume_latency_us", cpuSaveDir, cpu)
+			latencyOrig, err := ioutil.ReadFile(latencyFileOrig)
+			if err != nil {
+				if errors.Is(err, os.ErrNotExist) {
+					// the latency may have already been restored by a previous invocation of the hook
+					return nil
+				}
+				return err
+			}
+
+			// restore the original latency
+			latencyFile := fmt.Sprintf("%s/cpu%d/power/pm_qos_resume_latency_us", cpuDir, cpu)
```

Comment:
Somewhere above the string `"enable"` used once, and it is defined as `const`.

Here `pm_qos_resume_latency_us` is used 4 times, and it is not defined as const.

Reply:
In the case of pm_qos_resume_latency, since the usages are all in a single function are are mostly as part of a longer file path, I think that converting it to a constant is going to make the code less readable. I would prefer to leave this as is.